docs/DeveloperGuide.md

```diff
@@ -1,22 +1,145 @@
+- [JUnit 5](https://junit.org/junit5/) - Used for testing.
+- Code adapted from [AddressBook-Level3](https://github.com/se-edu/addressbook-level3).
+
+JavaNinja uses the following tools for development:
+- JUnit - Used for testing.
+- Gradle - Used for build automation.
 
 ## Design & implementation
+  
+### Design 
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+#### Architecture 
+
+Main components of the architecture:
+
+`Main` (consisting of class `JavaNinja`) is in charge of the app launch and shut down. 
+1. At app launch, it initialises the components in the correct order and connects them with each other
+2. At shut down, it shuts down the other components and invokes cleanup methods 
+
+The bulk of the app's work is done by the respective components: 
+1. `Cli`: It handles the user interface of the app 
+2. `Parser`: The command executor 
+3. `QuizManager`: Holds the data of the App in memory
+4. `Storage`: Reads data from, and writes data to, the hard disk.
+
+`Commons` represent a collection of classes used by other components.
+
+How the architecture components interact with each other: 
+
+The Sequence Diagram below shows how the components interact with each other for the scenario where the user issues the command `add Flashcard /q what's 2+3 /a 5`
+
+![image](https://github.com/user-attachments/assets/1dd443de-b538-4fb8-b4d1-3bff4643b3dd)
+
+
+### True/False Feature Implementation
+
+The `TrueFalse` class is responsible for handling true/false questions within the quiz application. This section describes its design, implementation details, and the rationale behind design choices.
+
+#### Feature Overview
+The `TrueFalse` class represents a true/false question. It stores the question text and the correct answer as a boolean. Additionally, it includes methods to validate user answers and display answer options.
+
+#### Implementation Details
+
+1. **Attributes**:
+   - `boolean correctAnswer`: Stores the correct answer for the question.
+
+2. **Constructor**:
+   - `TrueFalse(String questionText, boolean correctAnswer)`:
+     Initializes the question text and the correct answer. The constructor checks that `questionText` is non-null and non-empty, ensuring a valid question is always provided.
+
+3. **Methods**:
+   - `checkAnswer(String userAnswer)`:
+     This method converts the user's input (`"true"` or `"false"`) into a boolean and compares it to the correct answer. It throws an `IllegalArgumentException` for invalid inputs, ensuring only `"true"` or `"false"` strings are accepted.
+   - `printOptions()`:
+     Displays answer choices ("1. True" and "2. False") for the user. This makes it clear what options are available for true/false questions.
+   - `toString()`:
+     Returns a string representation of the question with a "(True/False)" label to indicate its type.
+
```

Comment:
Good work on giving the flow of events for the CheckAnswer feature, do add on the UML sequence diagram for that feature